src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java

```diff
@@ -106,26 +107,41 @@ public void sendPackets(
         {
             for (@NotNull final CitizenData citizen : citizens.values())
             {
-                if (citizen.isDirty() || hasNewSubscribers)
+                if (citizen.getCitizenEntity().isPresent())
                 {
-                    if (citizen.getCitizenEntity().isPresent())
+                    final List<EntityCitizen> list = colony.getWorld()
+                            .getEntities(EntityCitizen.class,
+                                    entityCitizen -> entityCitizen.getColony().getID() == colony.getID() && entityCitizen.getCitizenData().getId() == citizen.getId());
+
+                    if (!list.isEmpty() && citizen.getCitizenEntity().get().getEntityId() != list.get(0).getEntityId())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3655)